### PR TITLE
[Doc] Update `grants.md` references

### PR DESCRIPTION
**Pull request by emmanuel-ferdman**
## Changes
Commit fixes all the broken references in `grants.md`.

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: docs/resources/grants.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ resource "databricks_grants" "things" {
 
 ## Table grants
 
-You can grant `ALL_PRIVILEGES`, `APPLY_TAG`, `SELECT` and `MODIFY` privileges to [_`catalog.schema.table`_](tables.md) specified in the `table` attribute.
+You can grant `ALL_PRIVILEGES`, `APPLY_TAG`, `SELECT` and `MODIFY` privileges to [_`catalog.schema.table`_](../data-sources/tables.md) specified in the `table` attribute.
```
Comment (alexott) [inline docs/resources/grants.md:103]: it should be `sql_table.md` instead, not to data source.

File: docs/resources/grants.md [hunk 2]
```diff
@@ -138,7 +138,7 @@ resource "databricks_grants" "things" {
 
 ## View grants
 
-You can grant `ALL_PRIVILEGES`, `APPLY_TAG` and `SELECT` privileges to [_`catalog.schema.view`_](views.md) specified in `table` attribute.
+You can grant `ALL_PRIVILEGES`, `APPLY_TAG` and `SELECT` privileges to [_`catalog.schema.view`_](../data-sources/views.md) specified in `table` attribute.
```
Comment (alexott) [inline docs/resources/grants.md:141]: same here

File: docs/resources/grants.md [hunk 3]
```diff
@@ -172,7 +172,7 @@ resource "databricks_grants" "customers" {
 
 ## Volume grants
 
-You can grant `ALL_PRIVILEGES`, `READ_VOLUME` and `WRITE_VOLUME` privileges to [_`catalog.schema.volume`_](volumes.md) specified in the `volume` attribute.
+You can grant `ALL_PRIVILEGES`, `READ_VOLUME` and `WRITE_VOLUME` privileges to [_`catalog.schema.volume`_](../data-sources/volumes.md) specified in the `volume` attribute.
```
Comment (alexott) [inline docs/resources/grants.md:175]: this should be `volume.md` instead

**Review by alexott — CHANGES_REQUESTED** [review]

File: docs/resources/grants.md (hunk 2 above)
Comment (emmanuel-ferdman) [reply docs/resources/grants.md]: Should `catalog.schema.view` point to `sql_table.md` as well as `catalog.schema.table`? can't find `*view.md` in `resources` directory.

**Review by alexott — APPROVED** [review]

File: docs/resources/grants.md (hunk 2 above)
Comment (alexott) [reply docs/resources/grants.md]: yes, view is created via `sql_table`

**Pull request merged**